eventuals/concurrent.h

```diff
@@ -474,6 +483,32 @@ struct _Concurrent final {
     // start for each upstream value.
     template <typename E_>
     struct Fiber : TypeErasedFiber {
+      bool Reuse() override {
+        if (!done) {
+          return false;
+        }
+
+        CHECK(context.has_value());
+
+        if (context->in_use() || context->blocked()) {
+          return false;
+        }
+
+        done = false;
+
+        // Need to reinitialize the interrupt so that the
+        // previous eventual that registered with this
+        // interrupt won't get invoked as a handler!
+        interrupt.~Interrupt();
+        new (&interrupt) class Interrupt();
+
+        k.reset();
```

Comment:
Let's add a comment as to why we reset `k` before `context`, i.e., because `context` may be borrowed in `k`.